Makefile

```diff
@@ -88,6 +88,9 @@ gdbstub-test: $(BIN)
 	$(Q)tests/gdbstub.sh && $(call notice, [OK])
 endif
 
+# For tail-call elimination, we need a specific set of build flags applied.
+$(OUT)/emulate.o: CFLAGS += -fomit-frame-pointer -fno-stack-check -fno-stack-protector
```

Comment:
Add a comment:
> FIXME: On macOS + Apple Silicon, -fomit-frame-pointer might have a negative impact.

The option `-fomit-frame-pointer` would therefore be omitted on Arm64-based macOS.

Reply:
For `-fomit-frame-pointer`, see also https://github.com/godotengine/godot/issues/66296

Reply:
I test the GCC and clang with and without flag `-fno-omit-frame-pointer`, the result shows below.
| Microprocessor | compiler | without `-fomit-frame-pointer`| with `-fomit-frame-pointer` | 
|--------|-------|-------|-------|
| Core i7-8700   | clang-15 |        966.9798478         |         971.9516670         |
| Core i7-8700   | gcc-12   |        954.0062906         |         963.3369450         |
| eMag 8180      | clang-15 |       330.0058728        |         335.728476         |
| eMag 8180      | gcc-12   |      320.5225226         |   332.5611754|